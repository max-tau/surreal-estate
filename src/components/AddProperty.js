import React, { useState } from "react";
import "../styles/add-property.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      bedrooms: "",
      bathrooms: "",
      price: "",
      email: "",
      city: "Manchester",
      type: "Detached",
    },
  };
  const [fields, setFields] = useState(initialState.fields);
  const handleAddProperty = (e) => {
    e.preventDefault();
    console.log(fields);
  };
  const handleFieldChange = (e) => {
    setFields({ ...fields, [e.target.name]: e.target.value });
  };

  return (
    <>
      <h2 className="subheading">Add Property</h2>
      <div>
        <form className="add-property" onSubmit={handleAddProperty}>
          <label htmlFor="title">
            Title
            <input
              className="add-property-input"
              placeholder="Property tagline"
              type="text"
              id="title"
              name="title"
              value={fields.title}
              onChange={handleFieldChange}
            />
          </label>
          <label htmlFor="bedrooms">
            Bedrooms
            <input
              className="add-property-input"
              placeholder="Number of bedrooms"
              type="text"
              id="bedrooms"
              name="bedrooms"
              value={fields.bedrooms}
              onChange={handleFieldChange}
            />
          </label>
          <label htmlFor="bathrooms">
            Bathrooms
            <input
              className="add-property-input"
              placeholder="Number of bathrooms"
              type="text"
              id="bathrooms"
              name="bathrooms"
              value={fields.bathrooms}
              onChange={handleFieldChange}
            />
          </label>
          <label htmlFor="price">
            Price &#40;Â£&#41;
            <input
              className="add-property-input"
              placeholder="e.g. 100,000"
              type="number"
              min="0.00"
              step="250"
              id="price"
              name="price"
              value={fields.price}
              onChange={handleFieldChange}
            />
          </label>
          <label htmlFor="email">
            Email
            <input
              className="add-property-input"
              placeholder="name12@example.com"
              id="email"
              name="email"
              type="email"
              value={fields.email}
              onChange={handleFieldChange}
            />
          </label>
          <label htmlFor="city">
            City
            <select
              className="add-property-select"
              name="city"
              id="city"
              value={fields.city}
              onChange={handleFieldChange}
            >
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Sheffield">Sheffield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>
          <label htmlFor="type">
            Type
            <select
              className="add-property-select"
              name="type"
              id="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Flat">Flat</option>
              <option value="Bungalow">Bungalow</option>
            </select>
          </label>
          <button type="submit">Add</button>
        </form>
      </div>
    </>
  );
};

export default AddProperty;
