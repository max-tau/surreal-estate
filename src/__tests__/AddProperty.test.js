import React from "react";
import { render, screen } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  it("renders component correctly", () => {
    const { asFragment } = render(<AddProperty />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("contains a form with correct inputs and selects", () => {
    render(<AddProperty />);
    const textInputs = screen.queryAllByRole("textbox");
    const numberInput = screen.getByRole("spinbutton");
    const selectInputs = screen.queryAllByRole("combobox");

    expect(textInputs).toHaveLength(4);
    expect(numberInput).toBeInTheDocument();
    expect(selectInputs).toHaveLength(2);
  });
});
