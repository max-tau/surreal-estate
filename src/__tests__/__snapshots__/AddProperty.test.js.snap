// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders component correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    <h2
      class="add-property-subheading"
    >
      Add Property
    </h2>
    <form
      class="add-property-form"
    >
      <label
        for="title"
      >
        Title
        <input
          class="add-property-form-input"
          id="title"
          name="title"
          placeholder="Property tagline"
          type="text"
          value=""
        />
      </label>
      <label
        for="bedrooms"
      >
        Bedrooms
        <input
          class="add-property-form-input"
          id="bedrooms"
          name="bedrooms"
          placeholder="Number of bedrooms"
          step="1"
          type="number"
          value=""
        />
      </label>
      <label
        for="bathrooms"
      >
        Bathrooms
        <input
          class="add-property-form-input"
          id="bathrooms"
          name="bathrooms"
          placeholder="Number of bathrooms"
          step="1"
          type="number"
          value=""
        />
      </label>
      <label
        for="price"
      >
        Price (Â£)
        <input
          class="add-property-form-input"
          id="price"
          min="0.00"
          name="price"
          placeholder="e.g. 100,000"
          step="250"
          type="number"
          value=""
        />
      </label>
      <label
        for="email"
      >
        Email
        <input
          class="add-property-form-input"
          id="email"
          name="email"
          placeholder="name12@example.com"
          type="email"
          value=""
        />
      </label>
      <label
        for="city"
      >
        City
        <select
          class="add-property-form-select"
          id="city"
          name="city"
        >
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Sheffield"
          >
            Sheffield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </label>
      <label
        for="type"
      >
        Type
        <select
          class="add-property-form-select"
          id="type"
          name="type"
        >
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-detached
          </option>
          <option
            value="Terraced"
          >
            Terraced
          </option>
          <option
            value="End of Terrace"
          >
            End of Terrace
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Flat"
          >
            Flat
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
        </select>
      </label>
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
